pkg/server/image_pull.go

```diff
@@ -293,9 +293,7 @@ func (c *criService) registryHosts(auth *runtime.AuthConfig) docker.RegistryHost
 	return func(host string) ([]docker.RegistryHost, error) {
 		var registries []docker.RegistryHost
 
-		// Try mirrors in order, and then try the default registry if not tried.
-		endpoints, err := addDefaultEndpoint(
-			c.config.Registry.Mirrors[host].Endpoints, host)
+		endpoints, err := c.registryEndpoints(host)
 		if err != nil {
 			return nil, errors.Wrapf(err, "add default endpoint")
```

Comment:
needs a new err message for registry endpoints call vs add default endpoint...

Reply:
Done